chan.go

```diff
@@ -128,92 +161,94 @@ func (r *chanResponse) RawNext() (interface{}, error) {
 
 		return nil, io.EOF
 	case <-ctx.Done():
-		close(r.done)
 		return nil, ctx.Err()
 	}
 
 }
 
-type chanResponseEmitter struct {
-	ch   chan<- interface{}
-	wait chan struct{}
-	done <-chan struct{}
-
-	length *uint64
-	err    **cmdkit.Error
-
-	emitted bool
-}
+type chanResponseEmitter chanResponse
 
 func (re *chanResponseEmitter) Emit(v interface{}) error {
+	if e, ok := v.(*cmdkit.Error); ok {
```

Comment:
this is because we shouldnt be emitting errors ever anymore, right?

Reply:
Exactly. Should I remove this or are we not confident enough this doesn't happen? Or should I return that as an error, so it will definitely fail?

Reply:
i think we should keep it, and maybe even add a check for the non-pointerized version too just in case. Then also add a comment explaining why this bit of code is here.